pyupgrade.py

```diff
@@ -586,15 +589,18 @@ def _fix_format_literal(tokens: List[Token], end: int) -> None:
     last_int = -1
     for i in parts:
         try:
-            parsed = parse_format(tokens[i].src)
+            parsed = list(parse_format(tokens[i].src))
         except ValueError:
             # the format literal was malformed, skip it
             return
 
         # The last segment will always be the end of the string and not a
         # format, slice avoids the `None` format key
-        for _, fmtkey, spec, _ in parsed[:-1]:
-            if (
+        for i in range(len(parsed) - 1):
+            s, fmtkey, spec, _ = parsed[i]
+            if ENDS_ON_NAMED_UNICODE_RE.search(s):  # rebuild \N escape seqs
+                parsed[i] = (s + '{' + fmtkey + '}', None, None, None)
```

Comment:
this feels like a hack :/ -- I'd prefer if this lived in `parse_format` itself (would probably simplify some of the other cases as well?)